### DISCUSSION TITLE
'onDidChangeContent' bug in VSCode

### DISCUSSION THREAD

**Discussion by indrex** [community/community#49615] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

There seems to be a bug with Copilot extension in VSCode. It is preventing ipynb (jupyter) files from being saved. See the thead [here](https://github.com/microsoft/vscode-jupyter/issues/11435).

![image](https://user-images.githubusercontent.com/38977850/224018431-af58b814-4856-4380-9438-4913674fc525.png)

Edit: VSCode log as [gist](https://gist.github.com/indrex/aa9729a16ce429d2b390306202ebfec1)

**Comment by indrex** [comment]
I am pretty sure this is a Copilot related issue. How can we disable Copilot only in Jupyter notebook? The below doesn't seem to work. Copying from settings.json:

```
    "github.copilot.enable": {
        // "*": true,
        // disable for ipynb files
        "jupyter-notebook": false,
        "jupyter": false,
        "python": true
    },
```

Edit: I realize that only languages can be toggled, not the file extensions.

Edit: Switched to TabNine instead of Copilot. It is not as good, but Copilot is just not usable anymore in VSCode Jupyter.

**Comment by WXZhao7** [comment]
I aggre with @indrex. Copliot has broken my notebook and added many unattached `pykernel_launcher` processes. Now I have to disable Copilot when any notebook is open :(

**Comment by indrex** [comment, marked as answer]
Interim solution: install a month old version of both Jupyter and GitHub Copilot. It can be done using the drop down next to uninstall extension buttion in VSCode.

This resolves the bug.

**Reply by indrex** [reply]
I had raised a ticket with GitHub Copilot. They said they are taking a look.
